stock_partner_delivery_window/models/stock_picking.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+from odoo.tools.misc import format_datetime
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+    partner_delivery_window_warning = fields.Char(
+        compute="_compute_partner_delivery_window_warning"
+    )
+
+    def _planned_delivery_date(self):
+        return self.scheduled_date
+
+    @api.depends("partner_id", "scheduled_date")
+    def _compute_partner_delivery_window_warning(self):
+        for picking in self:
+            partner = picking.partner_id
+            picking.partner_delivery_window_warning = False
+            if not partner:
+                continue
+
+            anytime_delivery = partner and partner.delivery_time_preference == "anytime"
+            not_outgoing_picking = picking.picking_type_id.code != "outgoing"
+
+            if anytime_delivery or not_outgoing_picking:
+                continue
+
+            elif not partner.is_in_delivery_window(self._planned_delivery_date()):
+                self.partner_delivery_window_warning = (
+                    self._scheduled_date_no_delivery_window_match_msg()
+                )
```

Comment:
I tested on runboat, and with multiple delivery windows I get the following rendering:
![image](https://github.com/user-attachments/assets/9481a009-db22-45ca-a248-eca7a23a6507)

We should transform the `\n` carriage returns into HTML new lines, maybe with that helper (didn't try):
```suggestion
                picking.partner_delivery_window_warning = self.env["ir.qweb.field.text"].value_to_html(
                    picking._scheduled_date_no_delivery_window_match_msg(), {}
                )
```

We should replace `self` by `picking` too.

Reply:
Or simply by changing the `fields.Char` into a `fields.Text` for the warning msg?

Reply:
Done, thanks!

Reply:
Thank you, working fine now :)